infra/resources/modules/commons/apim.tf

```diff
@@ -52,6 +52,16 @@ module "apim" {
   tags = var.tags
 }
 
+resource "azurerm_key_vault_access_policy" "apim_policy" {
+  key_vault_id            = module.key_vault.id
+  tenant_id               = data.azurerm_client_config.current.tenant_id
+  object_id               = module.apim.principal_id
+  key_permissions         = []
+  secret_permissions      = ["Get"]
```

Comment:
**question**: do we need to grant the `List` permission too?

@Krusty93 I tried to do add (manually) a new named value, of type `key vault` on APIM, referencing the Trial System key vault, and the Azure portal is telling me "APIM must be granted List and Get secret management permission [...])". The portal proposes me to create them, and I canceled the operation (so I have not created any named value on the APIM).

I'm following [the steps explained in this StackOverflow answer](https://stackoverflow.com/questions/77499670/read-a-secret-key-from-azure-key-vault-in-apim/77500727#77500727) and it looks they did not grant the `List` permission.

What do you think about that? Should we try to give only the `Get` permission and check if it works?

Reply:
I double checked [Microsoft documentation](https://learn.microsoft.com/en-us/azure/api-management/api-management-howto-properties?tabs=azure-portal#:~:text=On%20the%20Permissions%20tab%2C%20under%20Secret%20permissions%2C%20select%20Get%20and%20List) and it looks the `List` permission is needed.

Do you think we should try using only the `Get` permission and then, if not working, adding the `List` or should I already add the `List` permission?

Reply:
Use both of them, it's not a problem at all.

By the way, @kin0992 do you have `List` access on the KeyVault? Because [the doc is saying](https://learn.microsoft.com/en-us/azure/api-management/api-management-howto-properties?tabs=azure-portal#add-a-key-vault-secret-to-api-management) that is mandatory to use Azure Portal (first blue box).

Reply:
@Krusty93 yep, I granted Get and List access to my user on the Azure Portal